core/src/main/scala/org/apache/spark/rdd/UnionRDD.scala

```diff
@@ -62,7 +64,21 @@ class UnionRDD[T: ClassTag](
     var rdds: Seq[RDD[T]])
   extends RDD[T](sc, Nil) {  // Nil since we implement getDependencies
 
+  // Evaluate partitions in parallel. Partitions of each rdd will be cached by the `partitions`
+  // val in `RDD`.
+  private lazy val evaluatePartitions: Unit = {
+    val threshold = conf.getInt("spark.rdd.parallelListingThreshold", 10)
+    if (threshold > 0 && rdds.length > threshold) {
+      val parArray = rdds.toParArray
+      parArray.tasksupport = new ForkJoinTaskSupport(new ForkJoinPool(threshold))
+      parArray.foreach(_.partitions)
+    } else {
```

Comment:
`false` and remove `rdds.foreach(_.partitions)` to make it evaluate in the original path. Thus, you can check if we go through this path in unit test.

Thanks.